File: core/src/main/java/com/lomekwi/cave/app/AppAudioOut.java
package com.lomekwi.cave.app;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.AudioDevice;

public class AppAudioOut {
    public static final int SAMPLE_RATE = 44100;
    private final AudioDevice audioDevice;

    public AppAudioOut() {
        audioDevice = Gdx.audio.newAudioDevice(SAMPLE_RATE, false);
    }

    public AudioDevice getAudioDevice() {
        return audioDevice;
    }
}

File: core/src/main/scala/com/lomekwi/cave/app/AppAudioOut.scala
package com.lomekwi.cave.app

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.audio.AudioDevice

class AppAudioOut {
  private final val audioDevice: AudioDevice = Gdx.audio.newAudioDevice(AppAudioOut.SAMPLE_RATE, false)

  def getAudioDevice(): AudioDevice = {
    audioDevice
  }

  def writeSamples(samples: Array[Float]): Unit = this.synchronized {
    audioDevice.writeSamples(samples, 0, samples.length)
  }
}

object AppAudioOut {
  final val SAMPLE_RATE = 44100
}
